bundles/org.openhab.core/src/main/java/org/openhab/core/util/Statistics.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.util;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * The {@link Statistics} is a class with statistics helper methods.
+ *
+ * @author Mark Herwege - Initial contribution
+ */
+@NonNullByDefault
+public class Statistics {
+
+    public static boolean randomQuickSelectSeed = false; // Can be enabled to always create a random pivot index for the
+                                                         // quickSelect algorithm, false will start from first value in
+                                                         // list as pivot and avoid random number generation
+
+    /**
+     * Find the median in a list of values
+     *
+     * @param inputList
+     * @return median of the values, null if the list is empty
+     */
+    public static @Nullable BigDecimal median(List<BigDecimal> inputList) {
+        ArrayList<BigDecimal> bdList = new ArrayList<>(inputList); // Make a copy that will get reordered
```

Comment:
Do you think we could skip this step, and document that this method will mutate the inputList? Then if the caller doesn't want its list mutated, the caller needs to create a copy of their list before passing to median()? This avoid having to allocate the list twice.

Reply:
Yes, we could. But it would also force us to check the provided inputList is mutable. Copying it makes sure of that. Also what happens if the input list is changed in another thread?
Copying it avoids all these problems. There is indeed a overhead, so it is worth considering, but I am not sure at the moment.